anoncreds/src/services/verifier.rs

```diff
@@ -370,26 +382,29 @@ fn compare_timestamps_from_proof_and_request(
 // which was added by the prover when creating `PresentCredentials`,
 // an arg for `create_presentation`.
 //
-// TODO: this timestamp should be compared with the provided interval
+// If the NRP is given at a timestamp outside of the request interval,
+// a wanring is provided for the verifier to decide if they will reject the proof or not
 fn validate_timestamp(
+    warnings: &mut Vec<String>,
     received_: &HashMap<String, Identifier>,
     referent: &str,
     global_interval: &Option<NonRevocedInterval>,
     local_interval: &Option<NonRevocedInterval>,
 ) -> Result<()> {
-    if get_non_revoc_interval(global_interval, local_interval).is_none() {
-        return Ok(());
-    }
-
-    if !received_
-        .get(referent)
-        .map(|attr| attr.timestamp.is_some())
-        .unwrap_or(false)
-    {
-        return Err(err_msg!("Missing timestamp"));
+    if let Some(interval) = get_non_revoc_interval(global_interval, local_interval) {
+        // If there is global or local interval, we compare the timestamp provided
+        if let Some(Some(ts)) = received_.get(referent).map(|attr| attr.timestamp) {
+            if ts.gt(&interval.to.unwrap_or(u64::MAX)) || ts.lt(&interval.from.unwrap_or(0)) {
+                // We add to NonRevProofWarnings that the referent is out of range
+                warnings.push(referent.to_string());
+            }
+            Ok(())
+        } else {
+            Err(err_msg!("Missing timestamp"))
+        }
+    } else {
+        Ok(())
```

Comment:
```suggestion
        } else {
            Err(err_msg!("Missing timestamp"))
        }
    Ok(())
```


That would work, right?